interpreter/cling/lib/Interpreter/ExternalInterpreterSource.cpp

```diff
@@ -0,0 +1,302 @@
+      }
+
+      // Put the name of the Decl imported with the
+      // DeclarationName coming from the parent, in  my map.
+      if (NamedDecl *fromNamedDecl = llvm::dyn_cast<NamedDecl>(From)) {
+        NamedDecl *toNamedDecl = llvm::dyn_cast<NamedDecl>(To);
+        m_Source.addToImportedDecls(toNamedDecl->getDeclName(),
+                                    fromNamedDecl->getDeclName());
+      }
+      if (DeclContext *fromDeclContext = llvm::dyn_cast<DeclContext>(From)) {
+        DeclContext *toDeclContext = llvm::dyn_cast<DeclContext>(To);
+        m_Source.addToImportedDeclContexts(toDeclContext, fromDeclContext);
+      }
+
+      return To;
+    }
+  };
+}
+
+namespace cling {
+
+  ExternalInterpreterSource::ExternalInterpreterSource(const cling::Interpreter *parent,
+                                              cling::Interpreter *child) :
+    m_ParentInterpreter(parent), m_ChildInterpreter(child) {
+
+    clang::DeclContext *parentTUDeclContext =
+      clang::TranslationUnitDecl::castToDeclContext(
+      m_ParentInterpreter->getCI()->getASTContext().getTranslationUnitDecl());
+
+    clang::DeclContext *childTUDeclContext =
+      clang::TranslationUnitDecl::castToDeclContext(
+      m_ChildInterpreter->getCI()->getASTContext().getTranslationUnitDecl());
+
+    // Also keep in the map of Decl Contexts the Translation Unit Decl Context
+    m_ImportedDeclContexts[childTUDeclContext] = parentTUDeclContext;
+  }
+
+  ClingASTImporter CreateClingASTImporter(ASTContext &toContext,
+                                          ASTContext &fromContext,
+                                          const Interpreter* child,
+                                          const Interpreter* parent,
+                                          ExternalInterpreterSource& source) {
+
+    FileManager &childFM = child->getCI()->getFileManager();
+    FileManager &parentFM = parent->getCI()->getFileManager();
+
+    return ClingASTImporter(toContext, childFM, fromContext, parentFM,
+                            /*MinimalImport : ON*/ true, source);
+  }
+
+  void ExternalInterpreterSource::ImportDecl(Decl *declToImport,
+                                   ASTImporter &importer,
+                                   DeclarationName &childDeclName,
+                                   DeclarationName &parentDeclName,
+                                   const DeclContext *childCurrentDeclContext) {
+
+    // Don't do the import if we have a Function Template.
+    // Not supported by clang.
+    // FIXME: This is also a temporary check. Will be de-activated
+    // once clang supports the import of function templates.
+    if (declToImport->isFunctionOrFunctionTemplate() 
+        && declToImport->isTemplateDecl())
+      return;
+
+    if (Decl *importedDecl = importer.Import(declToImport)) {
+      if (NamedDecl *importedNamedDecl = llvm::dyn_cast<NamedDecl>(importedDecl)) {
+        std::vector <NamedDecl*> declVector{importedNamedDecl};
```

Comment:
We don't need to build a `vector` anymore; `ArrayRef` accepts a single element these days. So this should work:
```
 SetExternalVisibleDeclsForName(childCurrentDeclContext,
                                        importedNamedDecl->getDeclName(),
                                        importedNamedDecl);
```